db/migrate/20160111013907_create_option_sets.rb

```diff
@@ -0,0 +1,11 @@
+class CreateOptionSets < ActiveRecord::Migration
+  def change
+    create_table :option_sets do |t|
+      t.references :division, index: true, null: false, foreign_key: true
+      t.string :model_type
```

Comment:
Is there a reason you aren't leveraging polymorphic associations here? You can see how that's done in the [media-attachable migration](madeline_system/db/migrate/20160106145757_create_media.rb). Should save you a lot of energy re-building methods rails provides in the concern.